validators/user.go

```diff
@@ -0,0 +1,139 @@
+package validators
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/somos831/somos-backend/models"
+)
+
+func (v *Validator) ValidateNewUser(newUser models.User) error {
+
+	if err := validateUserFields(newUser); err != nil {
+		return err
+	}
+
+	if newUser.Password == "" {
+		return errors.New("Password is a required field")
+	}
+
+	newUsername, err := v.isUniqueUsername(newUser.Username)
+	if err != nil {
+		return err
+	}
+
+	if !newUsername {
+		return errors.New("Username is already taken")
+	}
+
+	newAccount, err := v.isUniqueEmail(newUser.Email)
+	if err != nil {
+		return err
+	}
+
+	if !newAccount {
+		return errors.New("An account with the given email already exists")
+	}
+
+	return nil
+}
+
+func (v *Validator) ValidateUpdatedFields(user models.User) error {
+
+	if err := validateUserFields(user); err != nil {
+		return err
+	}
+
+	// Get current values: email and username
+	userRec, err := models.FindUserByID(v.DB, user.ID)
+	if err != nil {
+		return err
+	}
+
+	if userRec == nil {
+		return errors.New("user not found")
+	}
+
+	// Email is being updated
+	if userRec.Email != user.Email {
+		newEmail, err := v.isUniqueEmail(user.Email)
+
+		if err != nil {
+			return err
+		}
+
+		if !newEmail {
+			return errors.New("An account with the email provided already exists")
+		}
+	}
+
+	// Username is being updated
+	if userRec.Username != user.Username {
+		uniqueUsername, err := v.isUniqueUsername(user.Username)
+
+		if err != nil {
+			return err
+		}
+
+		if !uniqueUsername {
+			return errors.New("Username is already taken")
+		}
+	}
+
+	err = models.UpdateUser(v.DB, &user)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateUserFields(user models.User) error {
+
+	// Validate user input (e.g., check for required fields, format validation)
+	if user.Username == "" || user.Email == "" {
+		// http.Error(w, "Username, email, and password are required fields", http.StatusBadRequest)
```

Comment:
Let's remove commented code to avoid any confusion.

Reply:
Good catch, these slipped by. Thank you